### PR TITLE
Fix: Save dataset from simulation

### PR DESCRIPTION
**Pull request by Tom-Szendrey**
# Description
1) Move the save dataset from simulation into its own component
2) fix the java so this actually works

### CODE REVIEW

**Review by dvince2 — APPROVED** [review]

File: packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+		:disabled="isSaveDisabled"
+		outlined
+		title="Saves the current version of the model as a new Terarium asset"
+		@click="showSaveInput = !showSaveInput"
+	>
+		<span class="pi pi-save p-button-icon p-button-icon-left"></span>
+		<span class="p-button-text">Save as new dataset</span>
+	</Button>
+	<span v-if="showSaveInput" style="padding-left: 1em; padding-right: 2em">
+		<InputText v-model="saveAsName" class="post-fix" placeholder="New dataset name" />
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue:12]: Put styles in class for the span. Also this `post-fix` class isn't in the css, maybe you forgot to move it?

File: packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import { useProjects } from '@/composables/project';
+import InputText from 'primevue/inputtext';
+import Button from 'primevue/button';
+
+const props = defineProps<{
+	simulationRunId: string;
+}>();
+
+const saveAsName = ref('');
+const showSaveInput = ref(<boolean>false);
```
Comment (shawnyama) [inline packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue:39]: should be written as ref<boolean>(false) but just do ref(false) it already assumes its boolean

File: packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue (hunk 1 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue]: Thanks

File: packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue (hunk 2 above)
Comment (Tom-Szendrey) [reply packages/client/hmi-client/src/components/dataset/tera-save-dataset-from-simulation.vue]: Dont know why it was like that, good shout

**Pull request merged**